scripts/estimate-builds.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# pylint: disable=invalid-name
+
+import datetime
+from pathlib import Path
+import sys
+
+import croniter
+
+# Add the root of the repo to PYTHONPATH for utils
+ci_root = Path(__file__).resolve().parents[1]
+sys.path.insert(0, str(ci_root))
+# pylint: disable-next=wrong-import-position
+from utils import get_config_from_generator  # noqa: E402
+
+config = get_config_from_generator()
+
+builds_per_tree = {}
+for tree in config['tree_schedules']:
+    tree_name = tree['name']
+    tree_llvm_ver = tree['llvm_version']
+
+    # Calculate the number of times that a workflow runs in a week based on its
+    # schedule
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+    week_from_now = now + datetime.timedelta(weeks=1)
+    num_runs = len(
+        list(croniter.croniter_range(now, week_from_now, tree['schedule'])))
+
+    if tree_name not in builds_per_tree:
+        builds_per_tree[tree_name] = {'total': 0}
+    if tree_llvm_ver not in builds_per_tree[tree_name]:
+        builds_per_tree[tree_name][tree_llvm_ver] = 0
```

Comment:
I wonder if a `defaultdict` is more concise for these?
https://stackoverflow.com/a/9139961
It's nice when you have default values when a key does not exist.

Reply:
Yeah, I think that looks nicer: e500f351ecfae53c84cf87f6cae10388ef903f13